### PR TITLE
Add `square-*` code snippet/gist icons

### PR DESCRIPTION
**Pull request by danielbayley**
`symbol-snippet` from [codicons](https://microsoft.github.io/vscode-codicons/dist/codicon.html) and `code-square` (gist) from [octicons](https://primer.style/design/foundations/icons).

Related to #1266.

### CODE REVIEW

File: icons/square-dashed-bottom.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M3 17V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2v12" />
+  <path d="M4 21h1" />
+  <path d="M9 21h1" />
+  <path d="M14 21h1" />
+  <path d="M19 21h1" />
+</svg>
```
Comment (jguddas) [inline icons/square-dashed-bottom.svg:17]: What about something like this icon (48)?

```suggestion
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M5 21a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2" />
  <path d="M9 21h1" />
  <path d="M14 21h1" />
</svg>
```

**Comment by ericfennis** [comment]
Hmm not sure about these ones. The arrow brackets are a bit too small on 100% scale. Why should file code not work for this usecase?

**Comment by danielbayley** [comment]
> The arrow brackets are a bit too small on 100% scale.

They’re exactly the same size as https://lucide.dev/icons/search-code 🤷🏻‍♂️ going for consistency…

> Why should file code not work for this usecase?

@ericfennis Well a file and a [code] snippet are different things—a file can contain many code snippets (as demonstrated in #1359…)

As mentioned [above](https://github.com/lucide-icons/lucide/pull/1068#issue-1668146032), the main icons here have direct equivalents in Octicons and Codicons—which shows up in VS code like this:

Untitled
<br/>
<br/>

…and so is needed for the [vscode-lucide-icons](https://github.com/danielbayley/vscode-lucide-icons) package.

**Review by ericfennis — APPROVED** [review]
@danielbayley Thanks for your explanation. I approve this one!

**Pull request merged**